core/ide/che-core-ide-generators/src/main/java/org/eclipse/che/util/GeneratorUtils.java

```diff
@@ -41,6 +47,167 @@ public static String getClassFQN(String fileName, String content) throws IOExcep
+     *
+     * @param args
+     * @return File
+     */
+    public static File generateRootFolder(String[] args) {
+        String rootDirPath = ".";
+        // try to read argument
+        if (args.length == 1) {
+            if (args[0].startsWith(ROOT_DIR_PARAMETER)) {
+                rootDirPath = args[0].substring(ROOT_DIR_PARAMETER.length());
+            } else {
+                System.err.print("Wrong usage. There is only one allowed argument : "
+                        + ROOT_DIR_PARAMETER);//NOSONAR
+                System.exit(1);//NOSONAR
+            }
+        }
+
+        return new File(rootDirPath);
+    }
+
+    /**
+     * Generate to source of the Class
+     *
+     * @param rootFolder
+     * @param path
+     * @param fqn
+     * @throws IOException
+     */
+    public static void generateClassSource(File rootFolder, String path, Map<String, String> fqn) throws IOException {
+        File extManager = new File(rootFolder, path);
+        StringBuilder builder = new StringBuilder();
+        builder.append("package org.eclipse.che.ide.client;\n\n");
+        generateImports(builder);
+        generateClass(builder, path, fqn);
+        // flush content
+        FileUtils.writeStringToFile(extManager, builder.toString());
+    }
+
+    /**
+     * Generate imports.
+     *
+     * @param builder
+     */
+    public static void generateImports(StringBuilder builder) {
+        builder.append("import com.google.inject.Inject;\n");
+        builder.append("import com.google.inject.Provider;\n");
+        builder.append("import com.google.inject.Singleton;\n");
+        builder.append("import java.util.HashMap;\n");
+        builder.append("import java.util.Map;\n");
+    }
+
+    /**
+     * Generate Class declarations
+     *
+     * @param builder
+     * @param path
+     * @param fqn
+     */
+    public static void generateClass(StringBuilder builder, String path, Map<String, String> fqn) {
+        final String className = path.substring(path.lastIndexOf("/") + 1, path.indexOf("."));
+
+        String shortClassName = getShortClassName(className);
+
+        // generate class header
+        builder.append("/**\n");
+        builder.append(" * THIS CLASS WILL BE OVERRIDDEN BY MAVEN BUILD. DON'T EDIT CLASS, IT WILL HAVE NO EFFECT.\n");
+        builder.append(" */\n");
+        builder.append("@Singleton\n");
+        builder.append("@SuppressWarnings(\"rawtypes\")\n");
+        builder.append("public class ");
+        builder.append(className);
+        builder.append("\n");
+        builder.append("{\n");
+        builder.append("\n");
+
+        // field
+        builder.append(GeneratorUtils.TAB);
+        builder.append("/** Contains the map with all the ");
+        builder.append(shortClassName);
+        builder.append(" Providers <FullClassFQN, Provider>. */\n");
+        builder.append(GeneratorUtils.TAB);
+        builder.append("protected final Map<String, Provider> providers = new HashMap<>();\n\n");
```

Comment:
Please add TAB to the same line of code in situations when it won't make line too long because it improves readability